src/loader/typedtree_traverse.ml

```diff
@@ -0,0 +1,66 @@
+#if OCAML_VERSION >= (4, 14, 0)
```

Comment:
This can be done in a later PR, but why not using dune rules like this instead:
```lisp
(rule
  (enabled_if (< %{ocaml_version} 4.0.5))
  (action (copy file.404.ml file.ml)))

(rule
  (enabled_if (>= %{ocaml_version} 4.0.5))
  (action (copy file.405.ml file.ml)))
```

so we don't need the cppo syntax and we can apply ocamlformat on this file (and implementation.ml)

Reply:
I like the idea of a code without cppo though I think this wouldn't play well with Merlin. OCamlformat could perhaps support some form of cppo directives ? (eg. as structure items)